tracing-opentelemetry/src/layer.rs

```diff
@@ -192,9 +197,72 @@ impl<'a> field::Visit for SpanEventVisitor<'a> {
             }
         }
     }
+
+    /// Set attributes on the underlying OpenTelemetry [`Span`] using a [`std::error::Error`]'s
+    /// [`std::fmt::Display`] implementation. Also adds the `source` chain as an extra field
+    ///
+    /// [`Span`]: opentelemetry::trace::Span
+    fn record_error(
+        &mut self,
+        field: &tracing_core::Field,
+        value: &(dyn std::error::Error + 'static),
+    ) {
+        let mut chain = Vec::new();
+        let mut next_err = value.source();
+
+        while let Some(err) = next_err {
+            chain.push(Cow::Owned(err.to_string()));
+            next_err = err.source();
+        }
+
+        if self.2.record {
+            self.0
+                .attributes
+                .push(Key::new("exception.message").string(value.to_string()));
```

Comment:
nit, take it or leave it: i wonder if it's cheaper to call `value.to_string()` a single time, and then just clone the string? that way, we would still be allocating the new string and copying over all the bytes, but we wouldn't actually format the message multiple times, which might be _slightly_ less expensive...